geopmdpy/geopmdpy/gffi.py

```diff
@@ -44,21 +29,19 @@ def get_dl_geopm():
 
     '''
     global _dl_geopm
-    if type(_dl_geopm) is OSError:
-        raise _dl_geopm
+    if isinstance(_dl_geopm, Exception):
+        raise RuntimeError('Attempted to use libgeopm.so, which is not loaded. Make sure it is available '
+                           'in your system installs or in LD_LIBRARY_PATH') from _dl_geopm
     return _dl_geopm
 
-# Enforce load order of libgeopm.so and libgeopmd.so by loading
-# them together in this module.
```

Comment:
SInce load order is not required we could split this into two files, one in geopmdpy and one in geopmpy

Reply:
Opened https://github.com/geopm/geopm/issues/3624 so we can evaluate that once we've let this change soak in dev for a while.